src/java.base/share/classes/jdk/internal/module/ModuleInfoExtender.java

```diff
@@ -148,90 +150,40 @@ public void write(OutputStream out) throws IOException {
-                // drop existing attributes if there are replacements
-                if (name.equals(ClassFileConstants.MODULE_TARGET)
-                    && targetPlatform != null)
-                    return;
-                if (name.equals(ClassFileConstants.MODULE_RESOLUTION)
-                    && moduleResolution != null)
-                    return;
-                if (name.equals(ClassFileConstants.MODULE_HASHES)
-                    && hashes != null)
-                    return;
-
-                super.visitAttribute(attr);
 
+            // ModulePackages attribute
+            if (packages != null) {
+                List<PackageDesc> packageNames = packages.stream()
+                        .sorted()
+                        .map(s -> PackageDesc.of(s))
+                        .toList();
+                clb.with(ModulePackagesAttribute.ofNames(packageNames));
             }
-        };
-
-        List<Attribute> attrs = new ArrayList<>();
-        attrs.add(new ModuleTargetAttribute());
-        attrs.add(new ModuleResolutionAttribute());
-        attrs.add(new ModuleHashesAttribute());
-        cr.accept(cv, attrs.toArray(new Attribute[0]), 0);
-
-        // add ModuleTarget, ModuleResolution and ModuleHashes attributes
-        if (targetPlatform != null) {
-            cw.visitAttribute(new ModuleTargetAttribute(targetPlatform));
-        }
-        if (moduleResolution != null) {
-            int flags = moduleResolution.value();
-            cw.visitAttribute(new ModuleResolutionAttribute(flags));
-        }
-        if (hashes != null) {
-            String algorithm = hashes.algorithm();
-            List<String> names = new ArrayList<>();
-            List<byte[]> values = new ArrayList<>();
-            for (String name : hashes.names()) {
-                names.add(name);
-                values.add(hashes.hashFor(name));
-            }
-            cw.visitAttribute(new ModuleHashesAttribute(algorithm, names, values));
-        }
 
-        return cw.toByteArray();
+            // add ModuleTarget, ModuleResolution and ModuleHashes attributes
+            if (targetPlatform != null) {
+                clb.with(ModuleTargetAttribute.of(targetPlatform));
+            }
+            if (moduleResolution != null) {
+                clb.with(ModuleResolutionAttribute.of(moduleResolution.value()));
+            }
+            if (hashes != null) {
+                clb.with(ModuleHashesAttribute.of(hashes.algorithm(), hashes.hashes().entrySet().stream().map(he -> ModuleHashInfo.of(ModuleDesc.of(he.getKey()), he.getValue())).toList()));
+            }
+        }).andThen((clb, cle) -> {
+            if (v != null && cle instanceof ModuleAttribute ma) {
+                clb.with(ModuleAttribute.of(ma.moduleName(), ma.moduleFlagsMask(), clb.constantPool().utf8Entry(v.toString()), ma.requires(), ma.exports(), ma.opens(), ma.uses(), ma.provides()));
+            } else {
+                clb.accept(cle);
```

Comment:
Can you re-format this so that it's consistent with the existing code? Overfly long (180+ chars in this case) make it impossible to use side-by-side views when looking at changes.

Reply:
Fixed, I'm sorry I missed it.